mytest/models.py

```diff
@@ -44,6 +44,19 @@ def archive(self, *args, **kwargs) -> None:
         pass
 
 
+class SubFakeModel(FreezableFSMModelMixin):
+    FROZEN_IN_STATES = (
+        FakeStates.ACTIVE.value,
+        FakeStates.ARCHIVED.value,
+    )
+    FSM_STATE_FIELD_NAME = 'fake_model.state'
+    NON_FROZEN_FIELDS = ('fake_model.state', 'can_change_me')
+
+    fake_model = models.ForeignKey(FakeModel, on_delete=models.PROTECT)
```

Comment:
Just to understand a bit better the goal here :) So do you want the `child: SubFakeModel` to be frozen as long as the `parent: FakeModel` is the frozen state, right?  Meaning, the child itself actually doesn't get to say in which state(s) it will be frozen. It is merely depending on the parent's state?

Reply:
Sorry that I edited the message, and only afterward saw your reply. So it is the first case then.

-> Child to be frozen as long as the parent is frozen.

Reply:
As it is, the child obtain the state from the  parent and computes its own permission matrix.

Reply:
So you actually want the second one then :) Child obtain the parent's state, but use its own model's `FROZEN_IN_STATES` to decide frozen or not?

Reply:
Right it's like this, but I want to try otherwise https://github.com/ming-tung/django-fsm-freeze/pull/7#discussion_r691382480

Related:
https://github.com/ming-tung/django-fsm-freeze/pull/8